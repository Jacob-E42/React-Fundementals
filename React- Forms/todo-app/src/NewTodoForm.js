import React, { useState } from "react";
import { v4 as uuid } from "uuid";

const NewTodoForm = ({ add }) => {
	const [task, setTask] = useState("");
	const handleChange = e => {
		e.preventDefault();
		setTask(() => e.target.value);
	};
	const handleSubmit = e => {
		e.preventDefault();
		add({ task: task, id: uuid() });
		setTask("");
	};
	return (
		<form onSubmit={handleSubmit}>
			<label htmlFor="task">Task</label>
			<input
				type="text"
				id="task"
				name="task"
				value={task}
				onChange={handleChange}></input>
			<button type="submit">Submit</button>
		</form>
	);
};

export default NewTodoForm;
