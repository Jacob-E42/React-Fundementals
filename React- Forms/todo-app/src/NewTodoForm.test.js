import { render, fireEvent } from "@testing-library/react";
import NewTodoForm from "./NewTodoForm";

test("smoketest and snapshot test for NewTodoForm", () => {
	const { asFragment } = render(<NewTodoForm />);
	expect(asFragment).toMatchSnapshot();
});
