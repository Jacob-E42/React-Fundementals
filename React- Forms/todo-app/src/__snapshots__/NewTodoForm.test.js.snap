// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smoketest and snapshot test for NewTodoForm 1`] = `[Function]`;
